framework/test/gtest/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+#-----------------------------------------------------------------------------
+# Build and run the GTest Suite of tests
+#-----------------------------------------------------------------------------
+
+set(us_gtest_test_exe_name FrameworkTests)
+
+include_directories(${GTEST_INCLUDE_DIRS})
+
+#-----------------------------------------------------------------------------
+# Add test source files
+#-----------------------------------------------------------------------------
+set(_gtest_tests 
+  BundleVersionTest.cpp
+)
+
+add_executable(${us_gtest_test_exe_name} ${_gtest_tests})
+
+target_link_libraries(${us_gtest_test_exe_name} ${GTEST_BOTH_LIBRARIES})
+target_link_libraries(${us_gtest_test_exe_name} ${Framework_TARGET})
+
+# tests can be configured to run using add_test or GTEST_ADD_TEST (https://cmake.org/cmake/help/v3.0/module/FindGTest.html)
+# GTEST_ADD_TEST will run a seprate EXE for each test and test fixture in the EXE.
+# At this time, there isn't a need to use GTEST_ADD_TEST.
+
+# Run the GTest EXE from ctest.
+add_test(NAME ${us_gtest_test_exe_name} 
+  COMMAND ${us_gtest_test_exe_name}
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+)
+
+# Run the GTest EXE from valgrind
+if(US_MEMCHECK_COMMAND)
+  add_test(NAME memcheck_${us_gtest_test_exe_name} COMMAND ${US_MEMCHECK_COMMAND} --error-exitcode=1 ${US_RUNTIME_OUTPUT_DIRECTORY}/${us_gtest_test_exe_name}
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+  set_property(TEST memcheck_${us_gtest_test_exe_name} PROPERTY LABELS valgrind memcheck)
+endif()
+
+# Enable code coverage for GTest tests
+if(US_COVERAGE_COMMAND)
+  file(TO_NATIVE_PATH ${PROJECT_SOURCE_DIR} PROJECT_SOURCE_NATIVE_PATH)
+  add_test(NAME coverage_${us_gtest_test_exe_name} COMMAND ${US_COVERAGE_COMMAND} --verbose --cover_children --sources=${PROJECT_SOURCE_NATIVE_PATH} --export_type=binary:${us_gtest_test_exe_name}.cov --continue_after_cpp_exception -- $<TARGET_FILE:${us_gtest_test_exe_name}>
+      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+  set_property(TEST coverage_${us_gtest_test_exe_name} PROPERTY LABELS opencppcoverage)
+endif()
+
+# Add a Preprocessor flag if code coverage is turned ON
+if(US_ENABLE_COVERAGE)
+  target_compile_definitions(${us_gtest_test_exe_name} PUBLIC US_COVERAGE_ENABLED=1)
+endif()
+
+# Copy the Google Test libraries into the same folder as the
+# executable so that they can be seen at runtime on Windows.
+# Mac and Linux use RPATHs and do not need to do this.
+if (WIN32 AND US_USE_SYSTEM_GTEST)
+  foreach(lib_fullpath ${GTEST_BOTH_LIBRARIES})
+    get_filename_component(dir ${lib_fullpath} DIRECTORY)
+    get_filename_component(name_no_ext ${lib_fullpath} NAME_WE)
+    set(dll_file "${dir}/${name_no_ext}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+    add_custom_command(TARGET ${us_gtest_test_exe_name} POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different
+        "${dll_file}"
+        $<TARGET_FILE_DIR:${us_gtest_test_exe_name}>)
+  endforeach(lib_fullpath)
```

Comment:
What will happen is that the static libs will be copied, which is irrelevant in terms of running the EXE. At worst it copies the gtest static libs where they are not needed.

I can only speak to our use case, and copying the files is preferable because it doesn't pollute the ```PATH``` and keeps everything isolated. This is useful when doing large parallel builds of many different software projects.